src/main/scala/io/iohk/ethereum/vm/OpCode.scala

```diff
@@ -593,52 +595,67 @@ case object MSTORE8 extends OpCode(0x53, 2, 0, _.G_verylow) {
 
 case object SSTORE extends OpCode(0x55, 2, 0, _.G_zero) {
   protected def exec[W <: WorldStateProxy[W, S], S <: Storage[S]](state: ProgramState[W, S]): ProgramState[W, S] = {
-    val (Seq(offset, newValue), stack1) = state.stack.pop(2)
-    val currentValue = state.storage.load(offset)
-    val refund: BigInt = if (isEip1283Enabled(state)) {
-      val originalValue = state.originalWorld.getStorage(state.ownAddress).load(offset)
-      if (currentValue != newValue.toBigInt) {
-        if (originalValue == currentValue) { // fresh slot
-          if (originalValue != 0 && newValue.isZero)
-            state.config.feeSchedule.R_sclear
-          else 0
-        } else { // dirty slot
-          val clear = if (originalValue != 0) {
-            if (currentValue == 0)
-              -state.config.feeSchedule.R_sclear
-            else if (newValue.isZero)
-              state.config.feeSchedule.R_sclear
-            else
-              BigInt(0)
-          } else {
-            BigInt(0)
-          }
-
-          val reset = if (originalValue == newValue.toBigInt) {
-            if (UInt256(originalValue).isZero)
-              state.config.feeSchedule.R_sclear + state.config.feeSchedule.G_sreset - state.config.feeSchedule.G_sload
-            else
-              state.config.feeSchedule.G_sreset - state.config.feeSchedule.G_sload
+    val currentBlockNumber = state.env.blockHeader.number
+    val etcFork = state.config.blockchainConfig.etcForkForBlockNumber(currentBlockNumber)
+    val ethFork = state.config.blockchainConfig.ethForkForBlockNumber(currentBlockNumber)
+
+    val eip2200Enabled = isEip2200Enabled(etcFork, ethFork)
+    val eip1283Enabled = isEip1283Enabled(ethFork)
+
+    if(eip2200Enabled && state.gas <= state.config.feeSchedule.G_callstipend){
+        state.withError(OutOfGas)
```

Comment:
Usually `OutOfGas` errors zeros caller remaining gas like ` state.copy(gas = 0).withError(OutOfGas)`. Do you think we should try to do the same here ? (maybe it worth checking parity of geth)

Reply:
The tests from previous comment check that